src/solutions/LiquidityIncentive/components/SelectLiquidityPoolStep.tsx

```diff
@@ -0,0 +1,164 @@
+import {
+  FormControl,
+  FormErrorMessage,
+  FormHelperText,
+  FormLabel,
+  Input,
+  Skeleton,
+  Stack,
+  Text,
+} from "@chakra-ui/react"
+import { consts } from "@guildxyz/types"
+import Button from "components/common/Button"
+import { useCallback } from "react"
+import { useFormContext, useWatch } from "react-hook-form"
+import { usePairOfTokenId } from "requirements/Uniswap/hooks/usePairOfTokenId"
+import useParsePoolChain, {
+  UNISWAP_POOL_URL,
+  UniswapChains,
+} from "requirements/Uniswap/hooks/useParsePoolChain"
+import { useParsePoolTokenId } from "requirements/Uniswap/hooks/useParsePoolTokenId"
+import {
+  ADDRESS_REGEX,
+  useParseVaultAddress,
+} from "requirements/Uniswap/hooks/useParseVaultAddress"
+import { useSymbolsOfPair } from "requirements/Uniswap/hooks/useSymbolsOfPair"
+import { useTokenSymbolsOfPoolVault } from "requirements/Uniswap/hooks/useTokenSymbolsOfPoolVault"
+import ChainPicker from "requirements/common/ChainPicker"
+import parseFromObject from "utils/parseFromObject"
+import { CHAIN_CONFIG, Chains } from "wagmiConfig/chains"
+
+const SelectLiquidityPoolStep = ({ onContinue }: { onContinue: () => void }) => {
+  const {
+    formState: { errors, touchedFields },
+    setValue,
+    clearErrors,
+    register,
+  } = useFormContext()
+
+  const resetForm = () => {
+    if (!parseFromObject(touchedFields, "pool")?.address) return
```

Comment:
I think `address` is missing from the `LiquidityIncentiveForm` type & we don't really need to use the `parseFromObject` util here, we could just add a generic type to `useFormContext` & then use `touchedFields.pool?.address`. Same goes for the other `parseFromObject` calls in this file. We usually use that util when we need to parse some deep props from an object, or when we use a dynamic field name.